usersession/agent/rest_api.go

```diff
@@ -375,11 +398,23 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
 	}
 
 	// TODO: this message needs to be crafted better as it's the only thing guaranteed to be delivered.
-	summary := fmt.Sprintf(i18n.G("Update available for %s."), refreshInfo.InstanceName)
 	var urgencyLevel notification.Urgency
-	var body, icon string
+	var body, icon, name string
 	var hints []notification.Hint
 
+	name = refreshInfo.InstanceName
```

Comment:
instance name can contain the instance key, eg. `ohmygiraffe_beta`, but the `Name=..` in desktop file does not (or at least snapd does not add an instance key automatically). I suppose you could concatenate both to produce something reasonable, how about this:

- `ohmygiraffe` -> desktop file name `oh my giraffe`, presented name `oh my giraffe`
- `ohmygiraffe_beta` -> desktop file name `oh my giraffe`, presented name `oh my giraffe (beta)`
- `somesnap_beta` -> assume no name in desktop file, presented name `somesnap (beta)` 

you can use `snap.SplitInstanceName(name)` to get `<name>, <key>` tuple

Reply:
Mmm... good point...

Reply:
Done, but I don't know of any snap to make a manual test... Any suggestion?

Reply:
The problem I see is that maybe the string will be too long... the space in notifications is limited...